compile.c

```diff
@@ -4135,7 +4135,7 @@ compile_keyword_arg(rb_iseq_t *iseq, LINK_ANCHOR *const ret,
 
 	    *kw_arg_ptr = kw_arg;
 
-	    for (i=0; node != NULL; i++, node = node->nd_next->nd_next) {
+	    for (; node != NULL; i++, node = node->nd_next->nd_next) {
```

Comment:
I prefer to initialize `i=0` here because it is clear this loop start with `i=0`.
If you want to remove duplicated `i=0`, I prefer to remove first `int i=0` (only declare with `int i`).